index.js

```diff
@@ -138,6 +148,266 @@ app.intent('send.message - no', async conv => {
   conv.contexts.set('anything_else', 2);
 });
 
+/**
+ * join.conference
+ */
+app.intent('join.conference', async (conv, {target}) => {
+  const circuit = await getCircuit(conv);
+  if(!circuit){
+    return;
+  }
+
+  let matches = [];
+  let { inConferenceCall, conferenceAvailable, calls } = await getConferenceInfo(conv);
+
+  if (inConferenceCall) {
+    conv.ask("<speak>You're already in a conference call. Can I do anything else for you?</speak>");
+    conv.ask(new Suggestions('No, that\'s all', 'Yes'));
+    conv.contexts.set('anything_else', 2); 
+    return;
+  } 
+
+  if(!conferenceAvailable){
+    conv.ask(`<speak>No conferences available to join. Can I do something else for you?</speak>`);
+    conv.ask(new Suggestions('No, that\'s all', 'Yes'));
+    conv.contexts.set('anything_else', 2);
+    return;
+  }
+
+  // If target isn't undefined
+  if(target){
+    calls.forEach(c => {
+      if(c.title.toLowerCase() === target.toLowerCase()){
+        matches.push(c);
+      }
+    });
+
+    if (!matches.length) {
+      let prompt = `<speak>I cannot find any conference call with name ${target}. Here are the names of your current ongoing conferences. 
+      Which would you like to join?</speak>`;
+      let titles = calls.map((call) => call.title.toLowerCase());
+      conv.ask(prompt, new Suggestions(titles));
+      conv.contexts.set('joinconference_gettarget', 5, {
+        calls: calls
+      });
+      return;
+    }
+
+    if (matches.length === 1) {
+      // One result found. Ask user for confirmation.
+      const confId = matches[0].callId;
+      conv.ask(`<speak>Ready to join the <break time="0.5s"/>${target}<break time="0.5s"/> conference call?</speak>`, new Suggestions('Yes', `No, don't join`));
+      conv.contexts.set('joinconference_send', 5, {
+        confId: confId
+      });
+      return;
+    }
+
+    // Multiple matches. Tell user that they can either re-enter a conference or just join the first match.
+    conv.ask(`More than one conference found with the name ${target}. Joining the first ${target} conference.`);
+    const confId = matches[0].callId;
+    conv.ask(`<speak>Ready to join the <break time="0.5s"/>${target}<break time="0.5s"/> conference call?</speak>`, new Suggestions('Yes', `No, don't join`));
+    conv.contexts.set('joinconference_send', 5, {
+      confId: confId
+    });
+    return;
+  }
+
+  let titles = [];
+  calls.forEach((call) => {
+    if(call.title === ""){
+      let placeholder = truncate(call.topicPlaceholder);
+      titles.push(placeholder.toLowerCase());
```

Comment:
Why are you making these lowercase? Aren't they shown in suggestions?

Reply:
They are so this was a bad implementation. I've changed it so the suggestions are shown with whatever case they originally were and then only after they are converted to lowercase. They were being changed to lowercase so that the user can enter the title of the conference in any case they want and have it be a match. For example if the title of the conference was "Test" and the user entered "Join conference test" the target is now "test" and test !== Test so to normalize them i am checking against both of them lowercased.